scripts/tests/test_env_mutation_contract.py

```
@@ -0,0 +1,182 @@
+import unittest
+
+
+ROOT = pathlib.Path(__file__).resolve().parents[2]
+SCRIPT = ROOT / "scripts" / "check-env-mutation-contract.py"
+AUDITED_FILE = "crates/model-hf/src/store/local.rs"
+TODO = "// TODO: Audit that the environment access only happens in single-threaded code."
+
+
+class EnvironmentMutationContractTests(unittest.TestCase):
+    def test_repository_census_is_serialized_or_explicitly_deferred(self) -> None:
+        result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)